azkaban-web-server/src/test/java/azkaban/imagemgmt/services/ImageTypeServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+import azkaban.imagemgmt.daos.ImageTypeDaoImpl;
+import azkaban.imagemgmt.exeception.ImageMgmtValidationException;
+import azkaban.imagemgmt.models.ImageType;
+import azkaban.imagemgmt.models.RequestContext;
+import azkaban.imagemgmt.utils.JsonUtils;
+import java.io.IOException;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+public class ImageTypeServiceImplTest {
+  private ImageTypeDao imageTypeDao;
+  private ObjectMapper objectMapper;
+  private ImageTypeService imageTypeService;
+
+  @Before
+  public void setup() {
+    this.objectMapper = new ObjectMapper();
+    this.imageTypeDao = mock(ImageTypeDaoImpl.class);
+    this.imageTypeService = new ImageTypeServiceImpl(imageTypeDao, objectMapper);
+  }
+
+  @Test
+  public void testCreateImageType() throws Exception{
+    String jsonPayload = JsonUtils.readJsonAsString("image_management/image_type.json");
+    RequestContext requestContext = RequestContext.newBuilder()
+        .jsonPayload(jsonPayload)
+        .user("azkaban")
+        .build();
+    when(imageTypeDao.createImageType(any(ImageType.class))).thenReturn(100);
+    int imageTypeId = imageTypeService.createImageType(requestContext);
+    ArgumentCaptor<ImageType> imageTypeArgumentCaptor = ArgumentCaptor.forClass(ImageType.class);
+    verify(imageTypeDao, times(1)).createImageType(imageTypeArgumentCaptor.capture());
+    ImageType imageType = imageTypeArgumentCaptor.getValue();
+    Assert.assertEquals("kafka_push_job", imageType.getType());
+    Assert.assertEquals("azkaban", imageType.getCreatedBy());
+    Assert.assertEquals("image", imageType.getDeployable().getName());
+    Assert.assertNotNull(imageType.getOwnerships());
+    Assert.assertEquals(2, imageType.getOwnerships().size());
+    Assert.assertEquals(100, imageTypeId);
+  }
+
+  @Test
+  public void testCreateImageTypeForConfigs() throws Exception{
+    String jsonPayload = JsonUtils.readJsonAsString("image_management/image_type_configs.json");
+    RequestContext requestContext = RequestContext.newBuilder()
+        .jsonPayload(jsonPayload)
+        .user("azkaban")
+        .build();
+    when(imageTypeDao.createImageType(any(ImageType.class))).thenReturn(100);
+    int imageTypeId = imageTypeService.createImageType(requestContext);
+    ArgumentCaptor<ImageType> imageTypeArgumentCaptor = ArgumentCaptor.forClass(ImageType.class);
+    verify(imageTypeDao, times(1)).createImageType(imageTypeArgumentCaptor.capture());
+    ImageType imageType = imageTypeArgumentCaptor.getValue();
+    Assert.assertEquals("configs", imageType.getType());
+    Assert.assertEquals("azkaban", imageType.getCreatedBy());
+    Assert.assertEquals("tar", imageType.getDeployable().getName());
+    Assert.assertEquals(100, imageTypeId);
+  }
+
+  @Test(expected = ImageMgmtValidationException.class)
+  public void testCreateImageTypeInvalidType() throws IOException {
```

Comment:
Can you please add some failure cases like same image type name etc?

Reply:
Image type name is made unique in the database. Hence, duplicate validation is not done in the code. If duplicate image type name is provided, database will automatically throw exception. As the database is mocked it is not feasible to do this kind of validation. If we have in memory database for testing or test database this can be done easily. However, will check if this can be done on the mock instance. Will drop this if turns our to be complex.